modules/imgcodecs/include/opencv2/imgcodecs.hpp

```diff
@@ -332,6 +332,49 @@ CV_EXPORTS_W bool haveImageReader( const String& filename );
  */
 CV_EXPORTS_W bool haveImageWriter( const String& filename );
 
+/** @brief To read Multi Page images on demand
+
+The ImageCollection class provides iterator API to read multi page images on demand. Create iterator
+to the collection of the images and iterate over the collection. Decode the necessary page with operator*.
+
+The performance of page decoding is O(1) if collection is increment sequentially. If the user wants to access random page,
+then the time complexity is O(n) because the collection has to be reinitialized every time in order to go correct page.
+This is required because multipage codecs does not support going backwards.
+After decoding the one page, it is stored inside the collection cache. Hence, trying to get Mat object from already decoded page is O(1).
```

Comment:
I propose to make caching optional. It slows down some cases a bit, but may dramatically reduce memory consumption.

Reply:
@asmorkalov, this can be made optional very easily. Just call release/releaseCache() after you finished processing of each image. I think, it's quite simple and yet good enough workaround. I suggest not to make the API more complex at the moment

Reply:
Here is the example. Suppose the scenario is user wants to read images sequentially then discard them.

```cpp
ImageCollection collection("some.tiff");

int index = 0;
for(auto& page: collection) {
    // process page here
    collection.release(index);
    index++;
}
```

So the total memory usage here is actually `collection.size() * sizeof(cv::Mat (empty)) + sizeof(current page)`. If user release every image after processing,  there are no extra memory usage.

Reply:
Ok